src/foam/nanos/auth/ChangePasswordView.js

```diff
@@ -9,267 +9,166 @@ foam.CLASS({
-              }, function() {
-                this.addClass('logoCenterVertical');
-              })
-            .end()
-          .end()
-          .start()
-            .callIfElse( this.horizontal, function() {
-              this.addClass('horizontal');
-            }, function() {
-              this.addClass('centerVertical');
-            })
-            .start('h1').addClass('title-top').add(this.TITLE).end()
-            .start('p').addClass('subtitle').add(this.SUBTITLE).end()
-            .start(this.SectionView, {
-              data: this,
-              sectionName: 'resetPasswordSection'
-            }).addClass('contents').end()
-            .br().br()
-            .start().addClass('submitBtn')
-              .start(this.UPDATE_PASSWORD, { size: 'LARGE' }).end()
-              .start(this.RESET_PASSWORD, { size: 'LARGE' }).end()
-            .end()
-          .end();
-      }
-  ],
+      const self = this;
+      const logo = this.theme.largeLogo || this.theme.logo;
 
-  actions: [
-    {
-      name: 'resetPassword',
-      label: 'Confirm',
-      isAvailable: function() {
-        return !! this.token;
-      },
-      isEnabled: function(errors_) {
-        return ! errors_;
-      },
-      code: function(X) {
-        var user = this.User.create({
-          desiredPassword: this.newPassword
-        });
-        this.resetPasswordToken.processToken(null, user, this.token)
-        .then((_) => {
-          this.reset_();
-          this.stack.push({ class: 'foam.u2.view.LoginView', mode_: 'SignIn' }, this);
-          this.notify(this.SUCCESS_MSG);
-        }).catch((err) => {
-          this.notify(err.message, 'error');
-        });
-      }
-    },
-    {
-      name: 'updatePassword',
-      isAvailable: function() {
-        return ! this.token;
-      },
-      isEnabled: function(errors_) {
-        return ! errors_;
-      },
-      code: function(X) {
-        this.auth.updatePassword(null, this.originalPassword, this.newPassword)
-        .then((result) => {
-          this.user.copyFrom(result);
-          this.reset_();
-          this.notify(this.SUCCESS_MSG);
+      this.addClass(this.myClass())
+        // header
+        .callIf(this.showHeader, function() {
+          this.start().addClass(self.myClass('top-bar'))
+            .start('img').attr('src', logo).end()
+          .end();
         })
-        .catch((err) => {
-          this.notify(err.message, 'error');
-        });
-      }
+        // body
+        .start().addClass(this.myClass('content'))
+          // section
+          .start().addClass(this.myClass('section'))
+            .start(this.SectionView, {
+              data: this.passwordModel,
+              sectionName: this.passwordModel.cls_.model_.sections[0].name,
+              isCentered: true
+            }).end()
+          .end()
+          // link
+          .callIf(this.mode === 'send', function() {
+            this.start().addClass(self.myClass('link'))
```

Comment:
couldn't we just make them all go back to login? and get rid of this callif??

Reply:
For sending email and resetting password, it should be fine. When the user is updating their password , however, they need to be logged in so I thought it wouldn't be a good idea to redirect them to the login page after updating the page.